packages/starknet-snap/src/chain/transaction-service.ts

```diff
@@ -116,13 +115,13 @@ export class TransactionService {
       transactionsOnChain.push(tx);
       transactionsOnChainSet.add(tx.txnHash);
```

Comment:
if we do normalizeStarknetHex below, we should also do here to align the standard,

Reply:
I prefer Option 1, as it ensures that the transaction hash format is consistently normalized wherever it’s used. Since we pull data from the provider every time (and do not store it), normalizing at the source does not offer a significant advantage in terms of efficiency or maintainability.

Reply:
i think normalizing the source can be having a valid source of truth, as we also want the output to align as well